[2025-08-01 13:31:31,850] 12 utils.logging - INFO - downloading data from kaggle...
[2025-08-01 13:31:33,530] 16 utils.logging - INFO - saving raw data...
[2025-08-01 13:31:33,695] 20 utils.logging - INFO - splitting into train, test and validation
[2025-08-01 13:31:33,709] 23 utils.logging - INFO - saving train, test and validation
[2025-08-01 13:31:33,780] 20 utils.logging - INFO - Running preprocessing pipeline...
[2025-08-01 13:31:33,781] 80 utils.logging - INFO - Starting preprocessing pipeline...
[2025-08-01 13:31:33,781] 22 utils.logging - INFO - Reading training data for imbalance handling...
[2025-08-01 13:31:33,828] 27 utils.logging - INFO - Handling imbalance using RandomOverSampler...
[2025-08-01 13:31:33,839] 43 utils.logging - INFO - Building preprocessor pipeline...
[2025-08-01 13:31:33,839] 92 utils.logging - INFO - Fitting and transforming training data...
[2025-08-01 13:31:33,884] 97 utils.logging - INFO - Fitted preprocessor saved at model/preprocessor.pkl
[2025-08-01 13:31:33,983] 119 utils.logging - INFO - Saved preprocessed X to data/processed/X_processed.csv
[2025-08-01 13:31:33,983] 120 utils.logging - INFO - Saved target y to data/processed/y.csv
[2025-08-01 13:31:33,986] 23 utils.logging - INFO - Training model ....
[2025-08-01 13:31:33,986] 19 utils.logging - INFO - \U0001f4e5 Loading training data...
[2025-08-01 13:31:34,010] 23 utils.logging - INFO - \U0001f680 Starting MLflow run...
[2025-08-01 13:31:55,193] 46 utils.logging - INFO - \u2705 RandomForestClassifier trained and logged to MLflow.
[2025-08-01 13:31:55,199] 26 utils.logging - INFO - Evaluating the model
[2025-08-01 13:31:55,199] 12 utils.logging - INFO - \U0001f50d Loading validation data...
[2025-08-01 13:31:55,229] 17 utils.logging - INFO - \U0001f4e6 Loading preprocessor and transforming validation data...
[2025-08-01 13:31:55,279] 23 utils.logging - INFO - \U0001f4e5 Loading best model from MLflow...
[2025-08-01 13:31:55,279] 63 utils.logging - ERROR - \u274c Error occurred during evaluation.
Traceback (most recent call last):
  File "C:\Users\Hp\Desktop\CustomerChurn\src\components\evaluation.py", line 24, in evaluate_model
    logged_model_uri = f"runs:/{configs['best_run_id']}/model"
                                ~~~~~~~^^^^^^^^^^^^^^^
KeyError: 'best_run_id'
